### PR TITLE
chore(events): rule cannot have more than 5 targets

### PR DESCRIPTION
**Pull request by flexelem**
### Issue # (if applicable)

Closes #<issue number here>.

### Reason for this change
This is a minor refactor for preventing an Event Rule to have at most 5 associated targets defined during a synth. This is the main limit set by EventBridge https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-quota.html#eb-limits

### Description of changes
This is a very simple change in `aws-events` module to check if the `targets` array length of an `Event` is greater than 5. If it is then it throws an Error during build time.

### Description of how you validated changes
Added a unit test by in `rule.test.ts` by verifying an Error is being thrown when a Rule has more than 5 associated targets.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/aws-cdk-lib/aws-events/lib/rule.ts [hunk 1]
```diff
@@ -320,6 +320,8 @@ export class Rule extends Resource implements IRule {
   private renderTargets() {
     if (this.targets.length === 0) {
       return undefined;
+    } else if (this.targets.length > 5) {
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-events/lib/rule.ts:323]: What about adding the check inside the `validateRule` method instead?

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍 Left a suggestion

File: packages/aws-cdk-lib/aws-events/lib/rule.ts (hunk 1 above)
Comment (flexelem) [reply packages/aws-cdk-lib/aws-events/lib/rule.ts]: thanks makes sense. I moved it into `validateRule` method.

**Review by xazhao — APPROVED** [review]
Thanks for contributing. The change looks good to me.

**Pull request merged**